### PR TITLE
more linting

### PR DESCRIPTION
**Pull request by fperrad**
on demo & etc

### CODE REVIEW

File: demo/timing.c [hunk 1]
```diff
@@ -267,23 +267,23 @@ int main(void)
...
          }
-         mp_sub_d(&a, 1, &c);
+         mp_sub_d(&a, 1uL, &c);
          mp_mod(&b, &c, &b);
-         mp_set(&c, 3);
-         rr = 0;
-         tt = -1;
+         mp_set(&c, 3uL);
+         rr = 0u;
+         tt = 0xFFFFFFFFFFFFFFFFuLL;
```
Comment (sjaeckel) [inline demo/timing.c]: wouldn't `UINT64_MAX` be more appropriate?

File: demo/timing.c [hunk 2]
```diff
@@ -306,21 +306,21 @@ int main(void)
...
-         mp_add_d(&b, 1, &b);
+         mp_add_d(&b, 1uL, &b);
          mp_gcd(&a, &b, &c);
-      } while (mp_cmp_d(&c, 1) != MP_EQ);
+      } while (mp_cmp_d(&c, 1uL) != MP_EQ);
 
-      rr = 0;
-      tt = -1;
+      rr = 0u;
+      tt = 0xFFFFFFFFFFFFFFFFuLL;
```
Comment (sjaeckel) [inline demo/timing.c]: same

File: etc/mersenne.c [hunk 3]
```diff
@@ -68,10 +68,10 @@ static int is_mersenne(long s, int *pp)
    return res;
 }
 
-/* square root of a long < 65536 */
-static long i_sqrt(long x)
+/* square root of a int < 65536 */
```
Comment (sjaeckel) [inline etc/mersenne.c]: this comment is now lying for some architectures...

why even the change from `long` to `int`?

File: demo/timing.c [hunk 4]
```diff
@@ -183,15 +183,15 @@ int main(void)
          SLEEP;
          mp_rand(&a, cnt);
          mp_rand(&b, cnt);
-         rr = 0;
-         tt = -1;
+         rr = 0u;
+         tt = 0xFFFFFFFFFFFFFFFFuLL;
```
Comment (sjaeckel) [inline demo/timing.c]: same

File: demo/timing.c [hunk 5]
```diff
@@ -147,15 +147,15 @@ int main(void)
       SLEEP;
       mp_rand(&a, cnt);
       mp_rand(&b, cnt);
-      rr = 0;
-      tt = -1;
+      rr = 0u;
+      tt = 0xFFFFFFFFFFFFFFFFuLL;
```
Comment (sjaeckel) [inline demo/timing.c]: same

File: demo/timing.c [hunk 6]
```diff
@@ -126,15 +126,15 @@ int main(void)
...
```
Comment (sjaeckel) [inline demo/timing.c]: same

File: demo/timing.c (hunk 1 above)
Comment (fperrad) [reply demo/timing.c]: sure.

File: etc/mersenne.c (hunk 3 above)
Comment (fperrad) [reply etc/mersenne.c]: I'll revert this part.

File: demo/timing.c [hunk 7]
```diff
@@ -126,15 +126,15 @@ int main(void)
       SLEEP;
       mp_rand(&a, cnt);
       mp_rand(&b, cnt);
-      rr = 0;
-      tt = -1;
+      rr = 0u;
+      tt = UINT64_MAX ;
```
Comment (sjaeckel) [inline demo/timing.c]: \*plenk\* :)

doesn't your linter complain about that space before the `;`?

Comment (fperrad) [reply demo/timing.c]: @sjaeckel good catch

**Pull request merged**